Develop

joeflack4 opened the pull request:
See commits and files for itemized changes.

I think that the requirements.txt should definitely be added, though we haven't made use of Jinja2 yet.

The 'scan' module is based on the recent checks that were added to QTools2. I felt that it would be nice to have the ability to check for things before attempting a convert. I still think that is a good idea, however I am now not so sure about doing it in Python3 while the original code exists in Python2. Maybe better to create a multi-version compatible module. If preferred, I can remove those changes from this pull request.

jkpr commented:
@joeflack4 From my "scan" of your code, the only addition to Qtools2 is the `get_gi_integrigy` function? In any case, I think we can add this as part of `cascade.py`, and not duplicate so much code from Qtools2.

I agree with adding a requirements.txt file.

joeflack4 commented:
You're correct. Checking for several aspects of integrity with regards to geographic identifier data was something that I wanted to add. What I added was just a scaffold--not actually functional.

Everything else from 'scan' is basically taken from recent checks added to QTools. I converted to Python3 and added the checks to pmix as a 'scan' module, where these checks could be run independently of the conversion process. However I think it is a better idea to keep this code in QTools2 in Python2 since it exists there already.

I'll close this pull request, and include the requirements.txt as a separate pull request, or part of a PPP pull request.
